src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -857,6 +849,34 @@ public virtual HtmlString SetAntiforgeryCookieAndHeader()
             return HtmlString.Empty;
         }
 
+        public static bool ShouldRenderAttributeValue(object rawAttributeValue)
+        {
+            if (rawAttributeValue == null)
+            {
+                // Nothing to write
+                return false;
+            }
+
+            // Intentionally using is+cast here for performance reasons. This is more performant than as+bool?
+            // because of boxing.
```

Comment:
Comment applies in `GetStringAttributeValue()` as well.

More importantly, now doing the same casts and checks twice. Suggest instead extending `GetStringAttributeValue()` to return `null` when `rawAttributeValue` is `null` or a `false` Boolean value.